lib/compiler.js

```diff
@@ -65,6 +65,13 @@ exports.compile = function (code, options) {
     });
 
     importExportVisitor.finalizeHoisting();
+
+    if (getOption(options, "wrapWithRun")) {
+      result.code = "module.run(function(){" +
+        magicString.toString() + "//*/\n});";
+
+      return result;
+    }
```

Comment:
>Also there's no point in running module.runSetters() for the REPL module, since no other code will be importing anything from the REPL.

Ya, but `module.runSetters` doesn't hurt and toggling for repl ends up complicating several things for something that's a secondary, bonus, scenario. So what I'm saying is the REPL isn't worth the juggle cost. If it works great, but complicating the compiler, options, caching compiler, for it is a bit much.